include/core/io/platform/f4xx/ADCf4xx.hpp

```diff
@@ -34,14 +42,26 @@ class ADCf4xx : public ADC {
     /// supports multiple channels, so I made this one only use a single ADC.
     /// The F446re has 3 12 bit ADC's. Currently not able to use the other 2.
     /// The ADC will be initialized once then each channel will be added on.
-    static ADC_HandleTypeDef halADC;
+    static ADC_HandleTypeDef halADC[NUM_ADCS];
     /// Static list of all channels supported by the ADC
-    static Pin channels[MAX_CHANNELS];
-    /// Buffer for DMA where each spot represents the value read in from a
-    /// channel
-    static uint16_t buffer[MAX_CHANNELS];
-    static DMA_HandleTypeDef halDMA;
+    static Pin channels[NUM_ADCS][MAX_CHANNELS];
+    /// Buffer for DMA where each spot represents the value read in from a channel
+    static uint16_t buffer[NUM_ADCS][MAX_CHANNELS];
+    static DMA_HandleTypeDef halDMA[NUM_ADCS];
```

Comment:
I would suggest making a struct that holds all of this data then make an array of 3 structs for your ADCs. This way you index the array once at the beginning of a method using a pointer and the rest of the code focus on that single ADC.